visual-java/src/test/java/com/saucelabs/visual/integration/IntegrationBase.java

```diff
@@ -0,0 +1,155 @@
+package com.saucelabs.visual.integration;
+
+import com.saucelabs.visual.CheckOptions;
+import com.saucelabs.visual.VisualApi;
+import com.saucelabs.visual.graphql.type.DiffStatus;
+import com.saucelabs.visual.graphql.type.IgnoreSelectorIn;
+import com.saucelabs.visual.graphql.type.SelectorIn;
+import com.saucelabs.visual.graphql.type.SelectorType;
+import dev.failsafe.Failsafe;
+import dev.failsafe.RetryPolicy;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.time.Duration;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.openqa.selenium.By;
+import org.openqa.selenium.MutableCapabilities;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.firefox.FirefoxDriver;
+import org.openqa.selenium.remote.RemoteWebDriver;
+
+abstract class IntegrationBase {
+  private static final String username = System.getenv("SAUCE_USERNAME");
+  private static final String accessKey = System.getenv("SAUCE_ACCESS_KEY");
+  protected static RemoteWebDriver driver;
+  protected static VisualApi visual;
+  protected static boolean isLocal = Objects.equals(System.getenv("RUN_ON"), "local");
+
+  private static URL getDriverUrl() throws MalformedURLException {
+    if (username == null
+        || accessKey == null
+        || username.trim().isEmpty()
+        || accessKey.trim().isEmpty()) {
+      String err =
+          "Sauce Labs credentials not found. Please set SAUCE_USERNAME and SAUCE_ACCESS_KEY in your environment";
+      throw new RuntimeException(err);
+    }
+    String dataCenter = Optional.ofNullable(System.getenv("SAUCE_REGION")).orElse("us-west-1");
+    return new URL(
+        "https://"
+            + username
+            + ":"
+            + accessKey
+            + "@ondemand."
+            + (dataCenter.isEmpty() ? "us-west-1" : dataCenter)
+            + ".saucelabs.com:443/wd/hub");
+  }
+
+  @BeforeAll
+  public static void setUp() throws MalformedURLException {
+    MutableCapabilities caps = new MutableCapabilities();
+    caps.setCapability("browserName", "chrome");
+    caps.setCapability("browserVersion", "latest");
+    caps.setCapability("platformName", "Windows 10");
+    Map<String, Object> sauceOptions = new HashMap<>();
+    sauceOptions.put("screenResolution", "1600x1200");
+    caps.setCapability("sauce:options", sauceOptions);
+    driver = isLocal ? new FirefoxDriver() : new RemoteWebDriver(getDriverUrl(), caps);
```

Comment:
For `FirefoxDriver`, we should also make sure to set the screen resolution to 1600x1200.

Reply:
Screen resolution is only for the VM (Sauce) side, however, it does make sense to also set the window size which can be done for both so I shall add that

Reply:
Added setting of window size across both types. This will probably need some additional work to run local & remote tests side by side (since they would share snapshots now which would conflict and we'll need browsers on our Actions image), but this would atleast get us Sauce coverage for right now

Reply:
Why did you set it to something else than for Sauce?

Reply:
The `screenResolution` value is a sauce-specific value for the resolution of the entire _screen_, where setting the size of the window through selenium affects both local and remote executions so we can match them. It also generally has to be a smaller value than `screenResolution` since the window size needs to exclude any sort of OS toolbars.